### PR TITLE
[Fix] Fixes few jit and cuda errors in kornia.color

### PR DESCRIPTION
**Pull request by edgarriba**
### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/color/hsv.py [hunk 1]
```diff
@@ -27,7 +27,16 @@ def rgb_to_hsv(image: torch.Tensor) -> torch.Tensor:
...
                          .format(image.shape))
 
-    maxc, max_indices = image.max(-3)
+    # TODO: enable again for later versions than 1.6.0 or find a different implementation.
+    # It turns out that .max(...) does not return the index in the first position when
+    # all the inputs have the same value in CUDA.
+    # maxc, max_indices = image.max(-3)
+    if image.is_cuda and torch.__version__ == '1.6.0':
+        maxc, max_indices = image.cpu().max(-3)
+        maxc, max_indices = maxc.to(image), max_indices.to(image.device)
```
Comment (edgarriba) [inline kornia/color/hsv.py:36]: @shijianjian @pmeier @edgarriba @FGeri had to do this hack since somehow cuda .max have weird behaviour

File: kornia/color/hsv.py [hunk 2]
```diff
@@ -39,7 +48,10 @@ def rgb_to_hsv(image: torch.Tensor) -> torch.Tensor:
     deltac = torch.where(
         deltac == 0, torch.ones_like(deltac), deltac)
 
-    rc, gc, bc = torch.unbind(maxc.unsqueeze(-3) - image, dim=-3)
```
Comment (edgarriba) [inline kornia/color/hsv.py:42]: jit cuda doesn't like unbind in 1.6.0 but it does later :)

File: kornia/color/hsv.py (hunk 1 above)
Comment (pmeier) [reply kornia/color/hsv.py:36]: Why is this a requirement? Shouldn't this be irrelevant at which index the max value is?

Comment (edgarriba) [reply kornia/color/hsv.py:36]: this was an optimization done by @FGeri 
https://github.com/kornia/kornia/commit/56acb451088f67b0f00212f2e7d74df8eea189ee#diff-574268a035fea7cbc4fc3554acd1f0c40c6e8d42d2b7918298ef209f91f71dc9

**Review by ducha-aiki — APPROVED** [review]
